### PR TITLE
feat(headless): add moreResultsAvailable to ResultList state

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-525

### CODE REVIEW

**Review by jpmarceau — APPROVED** [review]
Looks good 👍 

@dmbrooke Maybe you're already aware of this, (or maybe I'm imagining 😅) but I think I remember you mentioning something about this a few days ago, so I'm tagging you just in case :)

File: packages/headless/src/controllers/result-list/headless-result-list.ts [hunk 1]
```diff
@@ -64,6 +67,12 @@ export interface ResultListState extends SearchStatusState {
    * The unique identifier of the last executed search.
    */
   searchUid: string;
+  /**
+   * Whether more results are available, using the same parameters as the last successful query.
+   *
+   * This property is not compatible with the `Pager` controller.
```
Comment (olamothe) [inline packages/headless/src/controllers/result-list/headless-result-list.ts:73]: Is it worth it to add more complete documentation/explanation about infinite scrolling/fetch more result/pager not being compatible/etc not being compatible with one another ?

This seems to be a point of confusion.

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/controllers/result-list/headless-result-list.ts (hunk 1 above)
Comment (kaitlynkatz) [reply packages/headless/src/controllers/result-list/headless-result-list.ts:73]: If I can jump in here, I think it's _definitely_ worth having more documentation on it, but I don't think this is the place to do it. We should have a page (or possibly multiple pages in a single section) on the documentation site about common use cases, and infinite scroll would be one of them. We do something similar for JSUI.

File: packages/headless/src/controllers/result-list/headless-result-list.test.ts [hunk 2]
```diff
@@ -105,5 +110,18 @@ describe('ResultList', () => {
...
+        engine.state.search.response.results.length;
+      await fetchMoreResultsAndWait(1, 0);
+
+      expect(mockDispatch).not.toHaveBeenCalled();
+      expect(engine.logger.info).toHaveBeenCalledWith(
+        'No more results are available for the result list to fetch.'
+      );
+      done();
+    });
   });
```
Comment (samisayegh) [inline packages/headless/src/controllers/result-list/headless-result-list.test.ts:126]: Could we add tests checking the value of the `moreResultsAvailable` state property when there are, and there are not, more results?

**Review by samisayegh — APPROVED** [review]

File: packages/headless/src/controllers/result-list/headless-result-list.ts (hunk 1 above)
Comment (ThibodeauJF) [reply packages/headless/src/controllers/result-list/headless-result-list.ts:73]: So, I remove these comments? 😄

Comment (kaitlynkatz) [reply packages/headless/src/controllers/result-list/headless-result-list.ts:73]: Nah, I think you should leave what you have. It'll be particularly helpful until we get around to writing those documentation site sections 👍🏻

File: packages/headless/src/controllers/result-list/headless-result-list.ts [hunk 3]
```diff
@@ -92,6 +101,10 @@ export function buildResultList(
     dispatch(registerFieldsToInclude(options.fieldsToInclude));
   }
 
+  const moreResultsAvailable = () =>
+    engine.state.search.response.results.length <
+    engine.state.search.response.totalCountFiltered;
```
Comment (btaillon) [inline packages/headless/src/controllers/result-list/headless-result-list.ts:106]: Does this work? Or do you mean: ```ts
const moreResultsAvailable = () =>
    engine.state.search.results.length <
    engine.state.search.response.totalCountFiltered;
```

Comment (ThibodeauJF) [reply packages/headless/src/controllers/result-list/headless-result-list.ts:106]: Yes you're right!

**Pull request merged**